added told-br (brazilian hate speech) dataset (#3683)
huggingface/datasets: datasets/told-br/README.md

---
annotations_creators:
- crowdsourced
language_creators:
- crowdsourced
languages:
- pt-BR
licenses:
- cc-by-sa-4.0
multilinguality:
- monolingual
pretty_name: ToLD-Br
size_categories:
- 10K<n<100K 
source_datasets:
- original
task_categories:
- text-classification
task_ids:

> - text-classification-other-hate-speech-detection

Reviewer: ```suggestion
- text-classification-other-hate-speech-detection
paperswithcode_id: told-br
```

- **Point of Contact:** joao.leite@estudante.ufscar.br

### Dataset Summary

ToLD-Br is the biggest dataset for toxic tweets in Brazilian Portuguese, crowdsourced by 42 annotators selected from a pool of 129 volunteers. Annotators were selected aiming to create a plural group in terms of demographics (ethnicity, sexual orientation, age, gender). Each tweet was labeled by three annotators in 6 possible categories: LGBTQ+phobia, Xenophobia, Obscene, Insult, Misogyny and Racism.

### Supported Tasks and Leaderboards

-`text-classification-other-hate-speech-detection`: The dataset can be used to train a model for Hate Speech Detection, either using it's multi-label classes or by grouping them into a binary Hate vs. Non-Hate class. A [BERT](https://huggingface.co/docs/transformers/model_doc/bert) model can be fine-tuned to perform this task and achieve 0.75 F1-Score for it's binary version.

### Languages

The text in the dataset is in Brazilian Portuguese, as spoken by Tweet users. The associated BCP-47 code is `pt-BR`.

## Dataset Structure

### Data Instances
ToLD-Br has two versions: binary and multilabel.  

> Multilabel:

Reviewer: ```suggestion

Multilabel:
```

A data point consists of the tweet text (string) followed by 6 categories that have values ranging from 0 to 3, meaning the amount of votes from annotators for that specific class on homophobia, obscene, insult, racism, misogyny and xenophobia.

An example from multilabel ToLD-Br looks as follows:

> {'text': '@user bandido dissimulado. esse sérgio moro é uma espécie de mal carater com ditadura e pitadas de atraso'
> 'homophobia': 0
> 'obscene': 0
> 'insult': 2
> 'racism': 0
> 'misogyny': 0
> 'xenophobia': 0}

Reviewer: ````suggestion
```
{'text': '@user bandido dissimulado. esse sérgio moro é uma espécie de mal carater com ditadura e pitadas de atraso'
'homophobia': 0
'obscene': 0
'insult': 2
'racism': 0
'misogyny': 0
'xenophobia': 0}
```
````

Binary:  
A data point consists of the tweet text (string) followed by a binary class "toxic" with values 0 or 1.

An example from binary ToLD-Br looks as follows:  

> {'text': '@user bandido dissimulado. esse sérgio moro é uma espécie de mal carater com ditadura e pitadas de atraso'
> 'toxic': 1}

Reviewer: ````suggestion
```
{'text': '@user bandido dissimulado. esse sérgio moro é uma espécie de mal carater com ditadura e pitadas de atraso'
'toxic': 1}
```
````

### Data Fields

Multilabel:  
- text: A string representing the tweet posted by a user. Mentions to other users are anonymized by replacing the mention with a @user tag.
- homophobia: numerical value {0, 1, 2, 3) representing the number of votes given by annotators flagging the respective tweet as homophobic.
- obscene: numerical value {0, 1, 2, 3) representing the number of votes given by annotators flagging the respective tweet as obscene.
- insult: numerical value {0, 1, 2, 3) representing the number of votes given by annotators flagging the respective tweet as insult.
- racism: numerical value {0, 1, 2, 3) representing the number of votes given by annotators flagging the respective tweet as racism.
- misogyny: numerical value {0, 1, 2, 3) representing the number of votes given by annotators flagging the respective tweet as misogyny.
- xenophobia: numerical value {0, 1, 2, 3) representing the number of votes given by annotators flagging the respective tweet as xenophobia.

Binary:  
- text: A string representing the tweet posted by a user. Mentions to other users are anonymized by replacing the mention with a @user tag.

> - toxic: numerical binary value {0, 1} representing if the respecive text is toxic/abusive or not. 

Reviewer: for consistency with the other binary classification datasets
```suggestion
- label: numerical binary value {0, 1} representing if the respective text is toxic/abusive or not. 
```
